edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs

```diff
@@ -280,6 +277,23 @@ await device.CloudConnection.Filter(cp => cp.IsActive)
             }
         }
 
+        async Task CloseAllCloudConnections()
+        {
+            using (await this.connectToCloudLock.LockAsync())
```

Comment:
We should log that we are closing all connections because device is disconnected

Reply:
You need changed to disconnected state callback